ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/conf/LogFeederMode.java

```diff
@@ -40,12 +40,16 @@ public String getText() {
   }
 
   public static LogFeederMode fromString(String text) {
-    for (LogFeederMode mode : LogFeederMode.values()) {
-      if (mode.text.equalsIgnoreCase(text)) {
-        return mode;
+    try {
+      for (LogFeederMode mode : LogFeederMode.values()) {
+        if (mode.text.equalsIgnoreCase(text)) {
+          return mode;
+        }
       }
+    } catch (IllegalArgumentException e) {
+      return LogFeederMode.DEFAULT;
```

Comment:
@rzuo  Can we add some logging here to indicate this configuration exception and the fallback to using the default configuration? This would help users troubleshoot issues more easily.